src/Chapter3.hs

```diff
@@ -1006,6 +1165,175 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+-- Data Types
+
+-- Actions - all possible actions
+data Action = AttackFoe | CastSpell Int | DrinkHealthPotion Int | RunAway
+                deriving (Show)
```

Comment:
To make it more typesafe, you could have two separate data types for `Knight` actions and `Monster` actions 🙂